### PR TITLE
Transaction signing payload is hashed first

### PR DESCRIPTION
**Pull request by gavofyork**
Instead of the encoded items that make up the signing payload, it's the Blake2/256 of the encoded items **whenever the payload length is greater than 256 bytes**. This keeps the data needed to craft a signature nice and short.

### CODE REVIEW

**Comment by gavofyork** [comment]
CC @jacogr

**Comment by rphmeier** [comment]
why? ed25519 signing is already hashing these items to the curve

**Comment by rphmeier** [comment]
also I think this will break some tests where things are signed manually

File: node/runtime/src/lib.rs [hunk 1]
```diff
@@ -66,8 +66,8 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("node"),
 	impl_name: create_runtime_str!("substrate-node"),
 	authoring_version: 10,
-	spec_version: 16,
-	impl_version: 16,
+	spec_version: 17,
```
Comment (jacogr) [inline node/runtime/src/lib.rs]: Just something to keep an eye on, it was bumped to 17 here - https://github.com/paritytech/substrate/pull/1591/files#diff-5e5e1c3aec9ddfde0a9054d062ab3db9R97 - so depending on which one goes in first, we may need an adjustment.

Comment (rphmeier) [reply node/runtime/src/lib.rs]: And in #1619 ... but we really should only need one bump per release at most.

Comment (gavofyork) [reply node/runtime/src/lib.rs]: Once people stop using `master` branch on public testnets, sure.

**Pull request merged**